src/helpers/augmentedReality/downloadObject.js

```diff
@@ -39,7 +39,22 @@ export const downloadObject = async ({ index, data, setData }) => {
 
       downloadedData[index].payload.downloadType = DOWNLOAD_TYPE.DOWNLOADED;
       downloadedData[index].payload.size += size;
-      downloadedData[index].payload.localUris.push({ uri, id, size, title, type });
+      downloadedData[index]?.payload?.localUris?.push({
```

Comment:
there is something set for `downloadedData[index].payload.size` in line 41 (and before)without checks and then in line 42 `downloadedData[index]?.payload?.localUris?` is checked for `push`. 
this is a bit confusing because if there is no `downloadedData[index].payload` in line 41 we already have an error and checks in line 42 do not help. i will refactor this with a surrounding check i guess, before setting any new values.